src/core/thread/mle.cpp

```diff
@@ -1607,7 +1607,8 @@ void Mle::HandleAttachTimer(void)
         // new parent candidate is compared with the current parent
         // and that it is indeed preferred over the current one.
 
-        if (mParentCandidate.GetState() == Neighbor::kStateParentResponse &&
+        if ((mParentCandidate.GetLinkInfo().GetLinkQuality() == 3 || mAttachState != kAttachStateParentRequestRouter) &&
```

Comment:
How about supplementing comments for this update to make it easier when reading this code in the future?